### DISCUSSION TITLE
Error when using the EnergyPlus 9.4 (python library mode) in macOS 11.4

### DISCUSSION THREAD

**Discussion by ghost** [NREL/EnergyPlus#8793] [Q&A] [answered]
I run the EnergyPlus API under the following conditions.
 - macOS 11.4 (apple scilicon)
 - python 3.8
 - EnergyPlus 9.4
 
The code I executed is as below.
```{.python}
import sys
sys.path.append("/Applications/EnergyPlus-9-4-0")
from pyenergyplus.api import EnergyPlusAPI
api = EnergyPlusAPI()
```

The following error message occurred.
```{.python}
api = EnergyPlusAPI()
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
~/Github/EPandPython/tmp.py in 
----> 9 api = EnergyPlusAPI()

/Applications/EnergyPlus-9-4-0/pyenergyplus/api.py in __init__(self, running_as_python_plugin)
     69                                          as this is already instantiated for Plugin workflows.
     70         """
---> 71         self.api = cdll.LoadLibrary(api_path())
     72         self.api.apiVersionFromEPlus.argtypes = [c_void_p]
     73         self.api.apiVersionFromEPlus.restype = c_char_p

~/miniforge3/envs/EPandPython/lib/python3.8/ctypes/__init__.py in LoadLibrary(self, name)
    449 
    450     def LoadLibrary(self, name):
--> 451         return self._dlltype(name)
    452 
    453 cdll = LibraryLoader(CDLL)

~/miniforge3/envs/EPandPython/lib/python3.8/ctypes/__init__.py in __init__(self, name, mode, handle, use_errno, use_last_error, winmode)
    371 
    372         if handle is None:
--> 373             self._handle = _dlopen(self._name, mode)
    374         else:
    375             self._handle = handle

OSError: dlopen(/Applications/EnergyPlus-9-4-0/libenergyplusapi.dylib, 6): no suitable image found.  Did find:
	/Applications/EnergyPlus-9-4-0/libenergyplusapi.dylib: mach-o, but wrong architecture
	/Applications/EnergyPlus-9-4-0/libenergyplusapi.dylib: stat() failed with errno=35
```

Does EnergyPlus API not support macOS yet?
I found the issues related to my question, but I do not know if it has improved in 9.4. ([Link 1](https://github.com/NREL/EnergyPlus/issues/7888), [Link 2](https://github.com/Myoldmopar/EPTravisTester/pull/4))

**Comment by Myoldmopar** [comment, marked as answer]
The error message appears to be pretty clear that you are trying to run a release on the wrong architecture.  EnergyPlus 9.4 was not released for m1 silicon.  We don't officially have an m1 release because we don't have a testing framework available to perform routine testing/development.  But one developer who does have an m1 system was kind enough to roll their own package release and post it to the 9.5 [release page](https://github.com/NREL/EnergyPlus/releases/tag/v9.5.0).  Try out 9.5 and see if it works better!  We routinely run the API tests on all three platforms (not m1), so hopefully it will work just fine for you.

**Reply by ghost** [reply]
Thanks. I'll try using 9.5.

**Reply by ghost** [reply]
I checked that it works well in version 9.5!

**Reply by Myoldmopar** [reply]
Great way to start the morning!

**Comment by akeomc** [comment]
I am getting a very similar error but am running python 3.9 and energyplus 22-2-0. Is there a mismatch between these architectures for a Mac M1 silicon? Thanks!

**Reply by asrjy** [reply]
Same here. Were you able to fix this?
